From fdc528306e4018b314f2efcb2be4e37672bddd75 Mon Sep 17 00:00:00 2001
From: Vadim Rutkovsky <vrutkovs@redhat.com>
Date: Thu, 9 Jul 2015 23:23:37 +0200
Subject: Adapt to new dirty tracking api

Mirrors commit b6d871bf2 (radeon: adopt for new X server dirty tracking
APIs.) in xf86-video-ati.

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=91284

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index c30cb3a..cd13820 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -729,7 +729,9 @@ drmmode_set_scanout_pixmap(xf86CrtcPtr crtc, PixmapPtr ppix)
 		screen->height = screenpix->drawable.height = max_height;
 	}
 	drmmode_crtc->scanout_pixmap_x = this_x;
-#ifdef HAS_DIRTYTRACKING2
+#ifdef HAS_DIRTYTRACKING_ROTATION
+	PixmapStartDirtyTracking(ppix, screenpix, 0, 0, this_x, 0, RR_Rotate_0);
+#elif defined(HAS_DIRTYTRACKING2)
 	PixmapStartDirtyTracking2(ppix, screenpix, 0, 0, this_x, 0);
 #else
 	PixmapStartDirtyTracking(ppix, screenpix, 0, 0);
diff --git a/src/nv_driver.c b/src/nv_driver.c
index 8e2ae03..4218e4f 100644
--- a/src/nv_driver.c
+++ b/src/nv_driver.c
@@ -557,7 +557,11 @@ redisplay_dirty(ScreenPtr screen, PixmapDirtyUpdatePtr dirty)
 	PixmapRegionInit(&pixregion, dirty->slave_dst);
 
 	DamageRegionAppend(&dirty->slave_dst->drawable, &pixregion);
+#ifdef HAS_DIRTYTRACKING_ROTATION
+	PixmapSyncDirtyHelper(dirty);
+#else
 	PixmapSyncDirtyHelper(dirty, &pixregion);
+#endif
 
 	DamageRegionProcessPending(&dirty->slave_dst->drawable);
 	RegionUninit(&pixregion);
-- 
cgit v0.10.2

